arrow/arrow.py

```diff
@@ -1524,7 +1688,9 @@ def _get_frames(cls, name):
         )
 
     @classmethod
-    def _get_iteration_params(cls, end, limit):
+    def _get_iteration_params(
+        cls, end: Any, limit: Union[int, None]
```

Comment:
`Union[int, None]` should be changed to `Optional[int]`.

Think it's worth refactoring this function to use `kwargs` or `args`, and adding two overloads, one with the end argument, and one with the limit argument? That way we enforce the fact that this should only take one or the other?

Reply:
Yeap we can achieve enforcing without using `kwargs` or `args`. but is it worth to refactor? I saw only one usage and even there the parameters are not manipulated value. They just pass from their arguments.

Reply:
Yeah it might not be worth the refactor. We have been looking to refactor span/interval for a while now, and when we do, I bet this function will be nuked anyways.